adapters/repos/db/queue/worker.go

```diff
@@ -0,0 +1,97 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package queue
+
+import (
+	"context"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+type Worker struct {
+	logger        logrus.FieldLogger
+	retryInterval time.Duration
+	ch            chan Batch
+}
+
+func NewWorker(logger logrus.FieldLogger, retryInterval time.Duration) (*Worker, chan Batch) {
+	ch := make(chan Batch)
+
+	return &Worker{
+		logger:        logger,
+		retryInterval: retryInterval,
+		ch:            ch,
+	}, ch
+}
+
+func (w *Worker) Run(ctx context.Context) {
```

Comment:
Run might return error

Reply:
It doesn't need to actually, it's a very simple function and it's spawned in a goroutine